### DISCUSSION TITLE
MRTrix2TrackVis :   error importing 'affine_from_fsl_mat_file' from 'dipy.tracking.utils'

### DISCUSSION THREAD

**Discussion by hollysully** [dipy/dipy#3147] [Tractography / Tractometry]
I am trying to use MRTrix2TrackVis (https://nipype.readthedocs.io/en/latest/api/generated/nipype.interfaces.mrtrix.convert.html)

When I run the lines of code below, I get this error when I run the last line:  ImportError: cannot import name 'affine_from_fsl_mat_file' from 'dipy.tracking.utils' (/opt/anaconda3/lib/python3.11/site-packages/dipy/tracking/utils.py)

```
import nipype.interfaces.mrtrix as mrt
tck2trk = mrt.MRTrix2TrackVis()
tck2trk.inputs.in_file = 'dwi_CSD_tracked.tck'
tck2trk.inputs.image_file = 'diffusion.nii'
tck2trk.run()    
```

The error seems to relate to what is discussed on this page (https://github.com/nipy/nipype/issues/3413), but I am not sure how to proceed

**Comment by skoudoro** [comment]
Hi @hollysully,

It has been a long time that this function has been removed from DIPY codebase. see https://github.com/dipy/dipy/pull/1939

I recommend you to ask the creator of this Nipype function to update the code directly in Nipype. I agree that it is related to https://github.com/nipy/nipype/issues/3413.

I hope it helps.

**Comment by hollysully** [comment]
Thank you! Could you recommend another method for extracting FA from an MRtrix .tck tract? Maybe a tutorial?

**Reply by arokem** [reply]
How about this? https://docs.dipy.org/stable/examples_built/streamline_analysis/afq_tract_profiles.html#sphx-glr-examples-built-streamline-analysis-afq-tract-profiles-py

**Comment by hollysully** [comment]
Thank you, Ariel! Do you also have a suggestion for segmenting the tract?

**Reply by skoudoro** [reply]
How about this command line: https://docs.dipy.org/stable/interfaces/bundle_segmentation_flow.html

**Reply by skoudoro** [reply]
this one can be helpful also: https://docs.dipy.org/stable/interfaces/buan_flow.html

**Comment by Garyfallidis** [comment]
The command line ``dipy_convert_tractogram`` was created to do such things @hollysully. Meaning convert tractograms. Then you can simply use an FA or any other image independently. They do not need to be in the same file.
